### ISSUE TITLE
Some groups operators should syntaxerror but do not

### ISSUE THREAD

**Issue by enebo** [ruby/prism#2073]
These should all throw SyntaxError.  Examples from ruby/spec:

```text
ruby -e "1 <=> 2 <=> 3"
ruby -e "1 == 2 == 3"
ruby -e "1 === 2 === 3"
ruby -e "1 != 2 != 3"
ruby -e "1 =~ 2 =~ 3"
ruby -e "1 !~ 2 !~ 3"
```

**Comment by kddnewton** [comment]
Ahh yeah this is a dup of https://github.com/ruby/prism/issues/1457. We have the mechanics to fix this now, but just haven't yet. It's on the list for soon though!

**Comment by makenowjust** [comment]
@kddnewton Why do you close this? I was surprised when I returned after implementing a fix (#2079) and it was closed (even though it wasn't fixed).

**Comment by kddnewton** [comment]
Oh sorry @makenowjust I closed it in favor of #1457 because I thought it was the same thing

**Comment by makenowjust** [comment]
Ah ok. Sorry.
